src/components/axes/timeAxis.ts

```diff
@@ -385,7 +385,7 @@ export module Axes {
         labelPos = tickPos;
       }
 
-      var tickLabels = container.selectAll("." + Axis.TICK_LABEL_CLASS).data(labelPos, (d) => d.valueOf());
+      var tickLabels = container.selectAll("." + Axis.TICK_LABEL_CLASS).data(labelPos, (d) => String(d.valueOf()));
```

Comment:
The keying function (the second argument to `.data()`) [has to return `string`s](https://github.com/mbostock/d3/wiki/Selections#data); previously the return value was implicitly being cast by Javascript.